### ISSUE TITLE
Cleaning chill temp and duration (cleanChillTempDuration)

### ISSUE THREAD

**Issue by DeirdreLoughnan** [lizzieinvancouver/egret#23]
@lizzieinvancouver I've temporarily added three columns for uncertainty (degC), temperature cycles for alternating temperatures (h), and daily light (h per day), then I'll tally them after to check if they're worth including. A sample entry would be

![Screenshot 2024-06-13 185946](https://github.com/lizzieinvancouver/egret/assets/167914871/bcb4a339-dca8-4377-a3c3-14cbc4ca492a)

for warm stratification of 20°C (16h dark) and 25°C (8h light) for 1 month followed by cold stratification of 2-4°C (dark) for 1 month.

Just a few notes:

- I changed some chill temperatures and durations altogether either because they're missing or they were actually germination temperatures, mostly in yang08, zhou08, and tylkowski21.
- exp1 in tylkowski21 is missing one data point (possibly because it was so similar to the one next to it), should I add it? All within the cleaning script of course.
- There were instances of germination within cold stratification with a warm + cold stratification e.g. Mamut20 and tylkowski21. I considered the chilling treatment to be the warm stratification, and the cold stratification to be the germination temperature, similar to what was done for Mamut20 (I only changed the chilling columns though).

Thank you @DeirdreLoughnan for the access.

_Originally posted by @kengi-neer in https://github.com/lizzieinvancouver/egret/issues/14#issuecomment-2167094617_

**Comment by DeirdreLoughnan** [comment]
@kengi-neer  posted the following in issue #14, and I have added some quick responses:

> Most of the unusual values have been cleaned already. 
Awesome work, thank you! 

>Some questions/notes:
> 1. Are the papers for bibby53, cousins10, and morozowska02 available? I haven't seen them in the drive and I didn't see them online through their titles.

Thanks for bringing this to my attention, I am not sure why some of these were scraped as @ngoj1 is correct that they are not in the source fine. I will look into it!

> 2. Data scraping for al-absi10/Al-Absi10 might have been done twice separately. yang18 seems to include all three papers, though all have different species.

Yes, the issue with Al-absi10 has been corrected in the mergeData.R file. But I will have to look into the yang18 issue.

> 3. Some points might be missing from pipinis12 and tylkowski91 since all others in the same table have been included.

If you think data was not entered, could you please add it, this might have been a simple mistake. 

> 4. Some response variable values are averaged over different treatments or interaction deemed not significant (chill temperature, duration, and even chemical) so they were not differentiated. I was thinking of just removing data for chill temperature and duration for those since they aren't helpful anyway. Examples in wytsalucy21 (chill temp - 4, 7, 10), Nin17 (chill dur - 15;30;60;90), okay11 (chill dur - 90, 120, 150), and tang21 (chill dur - 0, 15, 30, 60).

We don't want to remove any data at this stage of cleaning. It is ok to leave the values as they are for not, but please double check that there really is no way to differentiate the treatment levels in the figures. If we could fix this, then we should.

> 5. Quite many values for chill temperature and duration seem to be for germination, and vice versa, possibly storage too (will check yang18). Some examples are the corrected unusual values in yang08 (30/20) or even the corrected normal-looking values in tylkowski91 (3).
> 6. na11 has some contradicting descriptions of their treatments. Values stated in methods are different from those stated in results.

Interesting, let me think on these two.

> 7. Response variable and values for Schutz02 table 3 and pritchard93 table 3 seem to be mismatched.

Is this something that can be fixed by going back to the pdf, or do you think it is a typo in the publication?

> 8. Chill durations are rounded to the nearest integer.

The unit is in days, so this makes sense to me and seems fine.

> 9. Treatments without chilling in papers with chilling treatments were given a chill duration of 0 and chill temperature of NA. Or should I still include a chill temperature for treatments specified with a 0-week duration for grouping purposes?

Are these to indicate control treatments? I think for a control this sounds reasonable. 

> 10. Data points with chill temperature without duration in jacquemart21 and Naseri18.

I assume you have double checked the pdf's, if so having the duration be NA makes sense to me.

> I still have to check data with normal-looking chill temperature and duration values due to cases similar to 5 (yang08 and yang18 come to mind), but there are 2335, 875, and 1539 data points with temperature uncertainty, alternating temperature cycle, and daily light values. I put notes in the script, and separated and tagged changes by paper if needed.

Awesome thanks!

**Comment by DeirdreLoughnan** [comment]
@lizzieinvancouver, I spoke with @kengi-neer today, he has found several datapoints in papers that were not entered (I have found a few too). 

I know for OSPREE when we entered new raw data, we created new files and we tend not to edit the original excel files. Do you have a preference for what we do here? Should we just add new rows of data to the existing files (which were created by people no longer in the lab and might be simplest) or create a new file for "new" data entry?

**Comment by kengi-neer** [comment]
@lizzieinvancouver, some additional details on what @DeirdreLoughnan and I discussed earlier on some issues.

> Thanks for bringing this to my attention, I am not sure why some of these were scraped as @ngoj1 is correct that they are not in the source fine. I will look into it!

I found bibby53 listed in the source excel sheet from DM, which we think might have been supposed to be removed some time ago.

> We don't want to remove any data at this stage of cleaning. It is ok to leave the values as they are for not, but please double check that there really is no way to differentiate the treatment levels in the figures. If we could fix this, then we should.

Will not change for now, but maybe consider removing details on treatments since they are averaged across treatments such as different chill temperature or durations.

> Is this something that can be fixed by going back to the pdf, or do you think it is a typo in the publication?

I think they can be changed by going back to the pdf. For Schutz02, the response values are actually the number of days until
more than 1% of the seed had germinated, which could be moved to germ duration with the response values as 1 instead as I seem to have noticed to have been done for another paper for 50% germination (not sure, will try to find while checking them again). For pritchard93, the response values are actually the mean length, while the per germ columns are beside it, at the left for embryos isolated from seed and at the right for seeds.

> Are these to indicate control treatments? I think for a control this sounds reasonable.

Yes, I just have to think about controls for experiments wherein treatments are more of a list (e.g. soaking vs scarification vs warm strat vs cold strat) rather from a factorial arrangement (e.g. chemical x cold strat duration x photoperiod), which may be better off without a chill temperature at all.

> I assume you have double checked the pdf's, if so having the duration be NA makes sense to me.

For Naseri18, some seeds were simply left for more than 16 weeks without any specific duration. For jacquemart21, they simply mentioned cold stratification but without mention of any specific treatments (they have a lot of combination of stratification and chemical scarification treatments).

**Comment by lizzieinvancouver** [comment]
@kengi-neer Thanks for the update! These seem like good decisions and it's great to have them clearly documented in the git issue here.

**Comment by kengi-neer** [comment]
@lizzieinvancouver I would just like to follow up on how to add missing data points as @DeirdreLoughnan brought up last week.

Also, some notes/questions on cleaning:

1. For data averaged across treatments, usually due to different parameters not having significant interactions, I temporarily changed their values to "ave", please let me know if this is better than just putting it as NA.
2. Treatment for yang08 Table 3 were stated in different forms including "moist storage", "wet storage", and "stratification", with storage periods reaching up to 24 months but stated to have a similar procedure used for stratification. Should this be considered storage data or chill data?
4. yang16 and yang20 have two papers each as indicated in the data source file, but there are only data points for one paper in the data after merging (which is in the EGRET drive as yang16_2 and yang20). There could be missing data from yang16_1 Table 5 and yang20 Table 1, though yang20_1 is in another language except for the abstract.
5. The three yang18 papers all have storage data in chill columns.
6. yang18_2 seem to have mismatched figure/table sources; I'll clean it later after it has been fixed since it's easier to index by figure/table source.

**Comment by lizzieinvancouver** [comment]
> I know for OSPREE when we entered new raw data, we created new files and we tend not to edit the original excel files. Do you have a preference for what we do here? Should we just add new rows of data to the existing files (which were created by people no longer in the lab and might be simplest) or create a new file for "new" data entry?

@DeirdreLoughnan @kengi-neer I think creating a single file for new data entry (that various people can add to) sounds best for this.

**Comment by DeirdreLoughnan** [comment]
@lizzieinvancouver, @kengi-neer said:
>>I added the code for adding chill units in github ([here](https://github.com/lizzieinvancouver/egret/blob/main/analyses/cleaning/source/addChill.R)). The new columns are chillHours (chilling hours), chillPortions (chilling portions), and chillUnits (chilling units from Utah model). I broke down combined chilling treatments and got the sum of their values, which I think doesn't affect calculations for chilling hours and chilling units, but I am not sure for chilling portions.

Thanks @kengi-neer for working on this! I will take a look at the code and the resulting data and let you know if I have any questions.

**Comment by lizzieinvancouver** [comment]
@kengi-neer Thanks for your work on this! It all looks generally good, but we have (hopefully) finalized some [naming rules](https://github.com/lizzieinvancouver/egret/wiki/Cleaning-the-EGRET-data-overview#column-naming-conventions) and so I updated your code to these. Can you take a quick look at my changes (committed this afternoon) and make sure they seem okay?

**Comment by kengi-neer** [comment]
@lizzieinvancouver After looking at the commit history, the changes seem to be limited to the column names which were just modified according to the new naming conventions. I tried to make the script to not rely on the newly made columns, so they hopefully should not appear in any conditional statements. Assuming the changes were made using the find and replace tool in RStudio, the changes don't seem to affect the script's functionality except column names.

**Comment by lizzieinvancouver** [comment]
@kengi-neer That sounds good! I used find and replace in Sublime and all the code is running fine so I think we are good. Thanks for the second set of eyes to check this.

**Comment by lizzieinvancouver** [comment]
For posterity, changes are:
chill.tempCor -> chillTemp
chill.durationCor -> chillDuration
chill.tempUnc -> chillTempUnc 
chill.tempCycle -> chillTempCycle
chill.lightCycle -> chillLightCycle

**Comment by lizzieinvancouver** [comment]
@kengi-neer Here is some cleaning code I find around the cleaning/ folder from cleaning YEARS ago (so may be irrelevant now). Can you review it and confirm you caught all these issues in your cleaning code? Thank you!
```
# From clean_other.R, Original file called coordinate_cleaning_JS.R

unique(egret$chill.duration)
egret$chill.duration[which(egret$chill.duration == "unknown")] <- "NA"
#TO CHECK
# what does "90/30/90" mean - 90 days, then 30 days, then 90 days (3 different temperatures) - JS
#"Continuous cold stratification" kept in cold whole experiment? How long was experiment? - could not naseri18 paper on drive - JS
# is 0 a true zero? - there are 0 where they were controls (no chill)

chill9030 <- egret[which(egret$chill.duration == "90/30/90"),]
contcold <- egret[which(egret$chill.duration == "Continuous cold stratification"),]
chilldur0 <- egret[which(egret$chill.duration == "0"),]

# From cleaningDL.R
egret$chill.duration[which(egret$chill.duration == "unknown")] <- "NA"

```

Thank you!

**Comment by kengi-neer** [comment]
@lizzieinvancouver I checked the cleaning from the previous code and the "90/30/90" and "Continuous cold stratification" chill duration have already been cleaned. Most of the 0 chill temperature values in the cleaned column are actual 0°C chilling treatments, except for some in airi2009 which has now been cleaned.

**Comment by lizzieinvancouver** [comment]
@kengi-neer Thanks! I will start a new issue on the airi2009 problem.

**Comment by lizzieinvancouver** [comment]
@kengi-neer I am getting the following warnings when I run cleanChillTempDuration.R, can you check them? If you think you need them, let me know and we can consider adding a print statement to explain them. Thank you!

```
> source("cleaning/source/cleanChillTempDuration.R") # 50 columns
Warning messages:
1: In d$chillTemp[which(d$datasetID == "tylkowski91" & d$study == "exp1" &  :
  number of items to replace is not a multiple of replacement length
2: In d$chillDuration[which(d$datasetID == "tylkowski91" & d$study ==  :
  number of items to replace is not a multiple of replacement length
```

**Comment by kengi-neer** [comment]
@lizzieinvancouver I just resolved the warning.

**Comment by lizzieinvancouver** [comment]
@kengi-neer Could you help me write some notes here on what each column is? And give some examples of how to interpret the values? 

- chillTempUnc
- chillTempCycle
- chillLightCycle: Light cycle for chill duration (if mentioned) given in hours (what does 0 mean? Always dark? And what would "12 then 12 then 0" mean?) 

As a reminder, here's the unique values for each:
```

> unique(d$chillTempUnc)
 [1] NA           "1"          "1 then 1"   "2.5"        "0.5"        "0.45"       "0.6"        "NA then 1"  "1.5"       
[10] "NA and 1"   "2.5 then 1" "2"          "5"          "0.1"        "0.3"       
> unique(d$chillTempCycle)
 [1] NA                   "168 and 168"        "12 and 12"          "8 and 16"           "NA then NA then NA" "16 and 8"          
 [7] "16 and 8 then NA"   "24 and 24 then NA"  "24 then NA"         "24 and 24"          "1344 and 1344"      "672 and 672"       
[13] "336 and 336"        "NA then NA"        
> unique(d$chillLightCycle)
 [1] NA                  "0"                 "12 then 12 then 0" "8"                 "8 then 0"          "24"               
 [7] "0 then NA"         "8 then NA"         "0 and 24" 
```

**Comment by kengi-neer** [comment]
@lizzieinvancouver Please find the descriptions for these columns below:

- chillTempUnc (°C) - uncertainty in the chilling temperature values
Example: a chillTemp of 5 with chillTempUnc of 1 means a chill temp of 5°C ± 1°C or 4 °C -6 °C as stated in the article
- chillTempCycle (hours) - period of each alternating cycle for the entire chillDuration of that chilling condition
Example: a chillTemp of "10 and 20" with a chillTempCycle of "16 and 8" and a chillDuration of "14" means 16 hours at 10°C then 8 hours at 20°C and so on for 14 days; or, a chillTempCycle of "168 and 168" with a corresponding chillTemp of "4 and 22" and chillDuration of "112" means 168 hours (or 7 days) at 4°C then 168 hours at 22°C for 112 days
- chillLightCycle (hours/day) - daily light
Example: "12 then 12 then 0" means 12 hours of light for the first and second chilling conditions, and the third chilling condition is implemented in the dark

**Comment by lizzieinvancouver** [comment]
@kengi-neer This is great! I am working this and the storage issue so hoping I will get them sorted and closed soon.

**Comment by lizzieinvancouver** [comment]
@DeirdreLoughnan I get the feeling from my last comment that this should be tackled with cleanStorage .... can you see what you think? If you don't have time we could ask someone else to help.

**Comment by DeirdreLoughnan** [comment]
@lizzieinvancouver I have been working on the cleanChillTempDuration.R cleaning code, but it is taking awhile. I came across one paper that I think we should go back and maybe re-enter:

1. song20: I don't think the "storage temperature regime" in Fig 3A is entered properly. But this data might not be best suited to be entered in the storage column, as they also store seeds at -20 before this storage treatment is applied.

**Comment by DeirdreLoughnan** [comment]
@lizzieinvancouver I don't think the -20 should go in other treatments, I think the -20 is truly storage that occurred before the experiment began and then they exposed the seeds to different temperatures, which they do call a storage treatment. I would put this temperature regime into other treatment potentially or since it is more functionally akin to chilling treatments it could go there. Their "storage regime" last fairly short durations starting at 1 month.

I think Ken or one of us would be best to scrape it.

**Comment by lizzieinvancouver** [comment]
@DeirdreLoughnan Perhaps we should discuss this in person, as I think I understand it but it would be better to discuss while also discussing how we're entering storage and chilling ... and how many other storage treatment this papers has. For example, does it have storage AFTER the -20 storage treatment?

**Comment by kengi-neer** [comment]
@lizzieinvancouver @DeirdreLoughnan The "storage treatment" was applied after the cold stratification in the second experiment in song20. This is why I added it to the chilling columns instead of storage. However, the chilling treatments have to be corrected according to the cascading 1-month treatments applied. Replacing the code for song20 in cleanChillTempDuration.R with the code below will correct this. Also, I don't think we have scraped any data from Figures 3A, 4A, and 5A because most of the data are already in the other sub-figures. We are missing one data point by doing this, the last data point of panel A with the 5-month treatment with temperature regime of 5°C -> 10°C -> 15°C -> 20°C -> 25°C, each lasting for one month. Finally, data from Figures 4 and 5 were attributed to Figure 3.

```
temp <- c(rep("5", 5), rep("5 then 10", 4), rep("5 then 10 then 15", 3), rep("5 then 10 then 15 then 20", 2))
d$chillTemp[which(d$other.treatment == "cold stratification for 6 months")] <- paste0("NA then ", temp)
temp <- c("31", "62", "93", "124", "155", "31 then 31", "31 then 62", "31 then 93", "31 then 124", "31 then 31 then 31", "31 then 31 then 62", "31 then 31 then 93", "31 then 31 then 31 then 31", "31 then 31 then 31 then 62")
d$chillDuration[which(d$other.treatment == "cold stratification for 6 months")] <- paste0("186 then ", temp)
```

**Comment by lizzieinvancouver** [comment]
@DeirdreLoughnan I mentioned to @kengi-neer this today and he said he's happy to help but will wait to connect with you.

**Comment by DeirdreLoughnan** [comment]
@kengi-neer thanks for your help with this! 

If you have not done so already, could you add your above code to the cleaning script? I agree that we should not enter duplicated data if the same data is in multiple figures. If we are able to update the chilling treatment in the code and the other figures just have duplicated data, then I think this issue is fixed!

**Comment by kengi-neer** [comment]
@DeirdreLoughnan I have updated the chilling treatments now. Just to clarify, we don't have duplicated data for song20 right now. Actually, we are lacking one data point each from Figures 4 and 5.

**Comment by DeirdreLoughnan** [comment]
Awesome thanks! 

Oh I see, could you then make sure that datapoint gets scraped for both figures?

**Comment by lizzieinvancouver** [comment]
@kengi-neer Just checking in on this as we aim to finalize cleaning before the retreat next week if possible.

**Comment by lizzieinvancouver** [comment]
@kengi-neer No worries! But do you mean: (a) we need to scrape them (one data point in two figures, correct?) again, but you don't know how to use ImageJ/Fiji and so cannot scrape them or (b) something else? Then we can figure next steps.

**Comment by kengi-neer** [comment]
@lizzieinvancouver I meant (a) but the issue is we did not scrape data from first panel of Figures 3-5 because all other data can be found in other panels except for the 5°C -> 10°C -> 15°C -> 20°C -> 25°C treatment. But according to your reply in #70 and #71, cleaning out duplicates will be done later once response variables have been decided, so should we just scrape the first panel of Figures 3-5 which we know to have 4 duplicates and one unique data point each?

**Comment by lizzieinvancouver** [comment]
@christophe-rd Could you help re-scrape this data? I think I have over-assigned everyone else to tasks and I am hoping this will not be too much work, but let me know.

**Comment by christophe-rd** [comment]
@lizzieinvancouver yes I'll do it.

**Comment by christophe-rd** [comment]
@lizzieinvancouver @kengi-neer @DeirdreLoughnan 
I scraped panel a for figure 3, 4 and 5. Since they are temperature regimes, I input them as 5/10/15/20/25. Is that ok? 
Otherwise, I quickly checked if the figure scrapping for the other pannels made sens and it seemed like it did. 
I changed the figure names because all 3 were labeled as Figure 3. I also added the pannel letter (e.g. Figure 4b). I added the whole data for song20 in my[ scrapping csv file. ](https://github.com/lizzieinvancouver/egret/blob/b8d7eb2e927fdf15dd7c4aef2fc9b09cdb02bab3/data/scrapedEgret/egret_CRD.csv)
Let me know if I can do something else.

**Comment by lizzieinvancouver** [comment]
@christophe-rd Thank you! I'll leave this to @kengi-neer and @DeirdreLoughnan to review.

**Comment by vvandermeersch** [comment]
Might have spotted an issue with `airi09`.
From the paper: 
> For the cold stratification, seeds were kept at 4°C for 15, 30 and 60 days

But nothing is written in chillTemp and chillDuration for `treatment = stratification`.
I think there was a confusion between scarification and stratification, because it  is specified `scarif.type = cold`.

**Comment by DeirdreLoughnan** [comment]
Thanks @christophe-rd for re-scraping the data. 

I have updated cleanMisc.R to remove the original data. 

But as we are still coming across minor issues with this column, I am leaving this issue open for now.

**Comment by kengi-neer** [comment]
Notes:

- barnhill82: add chill temp of 3degC
- budisavljevic21: chill temp for table 2e should be 23degC
- dehgan84: missing stratification data as chilling though they are available in the paper

**Comment by kengi-neer** [comment]
The big chunk of code for cleaning Tables 3 and 4 of yang08 have been modified.

**Comment by christophe-rd** [comment]
@kengi-neer to confirm, should the remaining commented lines be deleted?

**Comment by christophe-rd** [comment]
> `airi09`

Fixed!

**Comment by christophe-rd** [comment]
> Notes:
> 
>     * barnhill82: add chill temp of 3degC
> 
>     * budisavljevic21: chill temp for table 2e should be 23degC
> 
>     * dehgan84: missing stratification data as chilling though they are available in the paper

Fixed! @kengi-neer whenever you have 5 minutes to check if we can delete the commented lines for yang08, it would be great!

**Comment by kengi-neer** [comment]
@christophe-rd Sorry for the late reply, I retained these commented lines and fixed the storage values over at cleanStorage.

**Comment by christophe-rd** [comment]
@kengi-neer thanks! 
I also fixed mamut20 and song20, but could you please double-check what I did to make sure I don't mess up your loops? No rush!

**Comment by kengi-neer** [comment]
@christophe-rd The code seems to be working fine. Some notes though:

- mamut20: the chilling conditions for germination percentages during stratification might need to be transferred to the germination conditions in Figure 3 
- song20: the chilling conditions for panels C-E of Figures 3-5 should follow those described in Figure 1 e.g. for panel D, chillTemp should be "5 then 10 then 15" and the chillDuration "31 then 30 then X" with X changed accordingly.

**Comment by lizzieinvancouver** [comment]
@DeirdreLoughnan is going to try to remember what we did with warm + cold strat treatments and pow-wow with @vvandermeersch about whether we're happy with decision rules.

**Comment by lizzieinvancouver** [comment]
@DeirdreLoughnan Did the pow-wow happen? I feel like it did and maybe I was even there ... but not sure.

**Comment by kengi-neer** [comment]
Chilling for grose57, Table 3, third column may need to include the germination conditions of the first column in its chilling columns as "17.2 then 4.4" with durations of "15 then *" with * as corresponding stratification durations.

**Comment by lizzieinvancouver** [comment]
> Chilling for grose57, Table 3, third column may need to include the germination conditions of the first column in its chilling columns as "17.2 then 4.4" with durations of "15 then *" with * as corresponding stratification durations.

@kengi-neer Is this a special case we should add to the script that calculates chilling? If so, could you help edit it?

**Comment by lizzieinvancouver** [comment]
Update on:

> @DeirdreLoughnan is going to try to remember what we did with warm + cold strat treatments and pow-wow with @vvandermeersch about whether we're happy with decision rules.

We discussed this (again) and we should see what happens in the code that @kengi-neer wrote when the LAST treatment (last after slashes) is warm.... @DeirdreLoughnan will check with @kengi-neer and ...

@vvandermeersch will check his treatment rules.

**Comment by lizzieinvancouver** [comment]
@christophe-rd will check all this work in this issue is done.

**Comment by kengi-neer** [comment]
@lizzieinvancouver I don't think grose57, Table 3, third column is a special case, code-wise. The chilling conditions just need to corrected for some entries.

For chilling calculations, I don't think having a warm stratification as the last treatment will derail any chilling calculations, but I could make some modifications according to new decisions. @DeirdreLoughnan Happy to discuss sometime.

**Comment by lizzieinvancouver** [comment]
@kengi-neer Thanks for the update! Could you correct the entries for grose57, Table 3, third column some time in the next 2-3 weeks? If you do not have time, I can definitely ask someone else.

**Comment by kengi-neer** [comment]
@lizzieinvancouver Will do, I will add an update here soon.

**Comment by lizzieinvancouver** [comment]
@kengi-neer Thanks!

**Comment by kengi-neer** [comment]
Some changes:

- modified chilling for grose57, Table 3, third column as mentioned above
- added chillTemp of 3 for barnhill82
- corrected chillTemp for warm stratification in budisavljevic21

Some things to resolve:

- chilling conditions for continuous cold stratification in naseri18 should be NA and transferred to germination conditions since they were never moved for incubation at a warm temperature
- chilling conditions for ren15 was put in storage conditions

**Comment by lizzieinvancouver** [comment]
Thanks @kengi-neer 

@DeirdreLoughnan Could you clean naseri18 in cleanGerminationTempDuration.r (#18)?

@vvandermeersch Could you clean ren15? I think it should be done in cleanChillTempDuration.R since cleanStorage comes after in the cleaning script (cleanall.R)?

**Comment by vvandermeersch** [comment]
@lizzieinvancouver @kengi-neer Here's the paragraph related to `exp2` of `ren2015`:

> #### Low temperature tolerance tests
> [...]
> In the laboratory experiment, the seeds of both species were stored in **controlled dry** and **moist soil** treatments (0% vs 30% soil moisture by mass, respectively). [...] Then, two groups of seed were stored hermetically at 4, −7, −15 and −20 °C. 

The idea of this experiment is to test for "_imbibitional chilling injury_". My guess is that _chilling_ conditions were entered as storage because of the dry/moist treatment. Moist conditions will be considered as chilling later in `combineStorageChill.R` (moist + cold), so I'm not sure it's an issue. 

However, I think there is another issue (more critical). The temperatures were entered as:
`table(d[d$datasetID == 'ren15' & d$study == "exp2", c("storageTemp")])`
| Temp (°C)    | # |
| -------- | ------- |
| 4  | 24  |
| -15 | 24  |
| -20    | 24    |
| -22    | 20    |
| -29    | 4    |

While the above paragraph only mentions 
> 4, −7, −15 and −20 °C

I wonder whether -22 and -29 should actually be -7 ?

**Comment by kengi-neer** [comment]
@vvandermeersch Yeah, sorry I forgot to mention it before. The response values for those data points match with the -7 treatment. As I understand, we consider all treatments termed "chilling" (or "chilling injury" as in this paper) into the chill columns despite dry/moist conditions. We only applied that consideration to storage conditions in #44, where we also found some dry chilling treatments. I'm not so sure how these were handled, @lizzieinvancouver @DeirdreLoughnan some clarification?

**Comment by lizzieinvancouver** [comment]
@vvandermeersch and @kengi-neer Thank you both for the quick and easy-to-follow work on this. I am pretty sure chilling conditions were supposed to be moist so I can see the issue here, however, we also try to match how authors present their studies and thus, I would say this should be entered as storage with moist/dry .... so perhaps just the number (C) is wrong? @vvandermeersch Did you do a quick CRTL + F for 22 and 29 to make sure there is not some additional detail the original data entry person was trying to capture? 


That said, am I imagining that we had a rule about chilling conditions @DeirdreLoughnan ? There's no mention of it in the meta tab of the egret file or in the Methods or [the scraping wiki](https://github.com/lizzieinvancouver/egret/wiki/Data-Scraping)

**Comment by vvandermeersch** [comment]
@lizzieinvancouver I checked and found nothing. The fact that -22 and -29 °C, taken together, represent `20 + 4 = 24` repetitions (as other temperatures, see table above) makes me quite confident that this is a scrapping error.

**Comment by lizzieinvancouver** [comment]
@vvandermeersch Great, thanks. Let's change to -7 and leave it in storage.

**Comment by vvandermeersch** [comment]
Done.

**Comment by lizzieinvancouver** [comment]
@vvandermeersch Thanks! I think this issue might be close to done ... waiting on (as best I can tell):

> @DeirdreLoughnan Could you clean naseri18 in cleanGerminationTempDuration.r (https://github.com/lizzieinvancouver/egret/issues/18)?

> @christophe-rd will check all this work in this issue is done.

If @christophe-rd @DeirdreLoughnan or @kengi-neer thinks I  missed something, please reply.

**Comment by DeirdreLoughnan** [comment]
@lizzieinvancouver @kengi-neer naseri18 has been fixed.

**Comment by lizzieinvancouver** [comment]
@DeirdreLoughnan Great! 
Now waiting on @christophe-rd (that is not to pressure you, just to make it easy for me to keep track of this issue).

**Comment by christophe-rd** [comment]
@lizzieinvancouver @DeirdreLoughnan 
I started looking at the code and found the following issues:

**airi2009**: can't find the paper in the dataset
**chien09**: can't find the paper in the dataset
**yang16**: can't find the paper in the dataset
**pritchard93**: which statement based on the treatment, but that specific treatment wasn't entered in either treatment or treatmentCor. @kengi-neer do you recall what you did there?
**battaglia97**: panel d of figure 1 if missing a lot of data. There should be roughly 80 data points and there are only 5 currently. At first glance, it seems like the other panels are also missing data points.

**Comment by kengi-neer** [comment]
@christophe-rd 

airi2009: changed to airi09 which was removed on Dec. 14 in cleanMisc.R. I am not sure though because I remember still seeing it sometime in January while cleaning the treatment column.
chien09: same paper but from different sources, chen08 was retained, notes in cleanMisc.R
yang16: renamed since there are 2 yang16 papers; changed to yang16_2 in cleandatasetID.R
pritchard93: This is supposed to correspond to Figure 2 which has since been removed on Feb. 12 in cleanMisc.R. This was supposed to change the inputted chill duration in weeks to days.
battaglia97: Maybe because only the final germination was scraped for Figure 1 Panels B-D. Not sure about Panel A though.

**Comment by christophe-rd** [comment]
@kengi-neer 
Ok thanks! For some reason, they were not coming up in my search!

**Comment by lizzieinvancouver** [comment]
@DeirdreLoughnan Says: chien09, yang16, pritchard93 ... she will check airi2009 and @christophe-rd will check if battaglia97 gets cleaned by end of cleanall.R... if not, suggest which script it should get cleaned in.

**Comment by DeirdreLoughnan** [comment]
Thanks @lizzieinvancouver, I have checked my notes and reviewed the data and the paper itself. I can't find a reason why it was removed, other than it being a mistake. I have removed this line of code and added the data back.

**Comment by christophe-rd** [comment]
@lizzieinvancouver @DeirdreLoughnan I looked up battaglia97 and here are the scripts that cleaned it:
clean.coordinates
cleanTreatments.R

However, neither addressed the fact that figure 1 only has a subset of the total data points in the figure pannels. I broke down the figure pannels below:
#### Fig1.A
Different periods of stratification, but it's missing a lot of data points. E.g. for 56 days, there are only 7 data points, but in the figure, there are 27. It's probably because it caps at 100% after a couple of days of incubation. Do we want all the points after it reaches a plateau? If so, we should probably rescrape that figure. 
#### Fig 1. B
It's the interaction of water stress X test temperature on one population
Do we want to keep that data? If so, we'll need to rescrape
#### Fig 1. C
It's the interaction of water stress on a single on another population.
Same question as Fig 1. B
#### Fig 1. D
Each slope represents five source population after 14 days of stratification. Currently, there is only 5 data points, so it's missing the vast majority of the data points of that figure.

Should we rescrape it? If so I could do it.

**Comment by lizzieinvancouver** [comment]
@christophe-rd Thanks for keeping at this! I think we should just rescrape it (assuming you checked and it is not present in the raw data). 

You could wait and train Devina on scraping data in May or do it yourself. Doing it yourself will likely be faster than training and then checking Devina's work so I would only suggest you train Devina if you think we will find a lot more stuff to scrape (thoughts @DeirdreLoughnan ?).

**Comment by DeirdreLoughnan** [comment]
@christophe-rd @lizzieinvancouver I agree, it sounds like this is useful data and worth re-scraping it. I also agree that teaching someone new how to scrape data takes a long time! It feels to me like we are close to the end, so it is probably not worth it.

**Comment by lizzieinvancouver** [comment]
> It feels to me like we are close to the end, so it is probably not worth it.

@DeirdreLoughnan I agree, so let's not train Devina.

**Comment by lizzieinvancouver** [comment]
@lizzieinvancouver will read all of this and hopefully close! See also wiki notes from 7 May meeting.

**Comment by christophe-rd** [comment]
@lizzieinvancouver I needed to re-scrape battaglia97, which I just did. However, I excluded fig 1 panels b and c for now as it was for the interaction of water stress and temperature. Is that ok? If so, I pushed my csv with the newly scraped data.

**Comment by lizzieinvancouver** [comment]
I would be ready to close this but @christophe-rd Can you confirm you finished scraping the interactions in battaglia97? See comment JUST above. 

And ... @vvandermeersch and @DeirdreLoughnan, can you confirm we did these things below (from March 19 by me):

> We discussed this (again) and we should see what happens in the code that @kengi-neer wrote when the LAST treatment (last after slashes) is warm.... @DeirdreLoughnan will check with @kengi-neer and ...
> 
> @vvandermeersch will check his treatment rules.

**Comment by christophe-rd** [comment]
@lizzieinvancouver Just scraped it battaglia97! All 4 figures are now scraped and pushed the edits!

**Comment by lizzieinvancouver** [comment]
@DeirdreLoughnan @vvandermeersch I am waiting on replies to the above so I can close this issue.
